src/libos/crates/jindisk/src/util/disk_array.rs

```diff
@@ -79,9 +84,9 @@ impl<T: Serialize> DiskArray<T> {
     }
 
     pub async fn persist(&self, _root_key: &Key) -> Result<()> {
-        // TODO: Add encryption(symmetric) for blocks
         for (hba, block) in self.table.iter() {
-            self.disk.write(*hba, block.as_slice()).await?;
+            let ciphertext = DefaultCryptor::symm_encrypt_block(block.as_slice(), &self.key)?;
```

Comment:
Is there security weakness in using the same key for every block? I thought the crypto API should take an argument for user to decide the IV. This way, different block can be encrypted with different IV, although the key is the same. Should our crypto API allow providing IVs?